icu4c/source/common/uresbund.cpp

```diff
@@ -3265,18 +3269,18 @@ ures_getFunctionalEquivalent(char *result, int32_t resultCapacity,
 
         getParentForFunctionalEquivalent(found,res,&bund1,parent,1023);
         ures_close(res);
+        subStatus = U_ZERO_ERROR;
     } while(!full[0] && *found && U_SUCCESS(*status));
 
     if((full[0]==0) && kwVal != defVal) {
 #if defined(URES_TREE_DEBUG)
         fprintf(stderr, "Failed to locate kw %s - try default %s\n", kwVal.data(), defVal);
 #endif
         kwVal.clear().append(defVal, subStatus);
-        uprv_strcpy(parent, base);
-        uprv_strcpy(found, base);
-        
+        base.extract(parent, UPRV_LENGTHOF(parent), subStatus);
+        base.extract(found, UPRV_LENGTHOF(found), subStatus);
```

Comment:
This cause https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=65635 because found is not always null terminiated